work_queue/src/work_queue.c

```diff
@@ -914,6 +956,16 @@ void update_read_catalog_factory(struct work_queue *q, time_t stoptime) {
 	}
 
 	jx_delete(jexpr);
+
+	// Remove outdated factories
+	hash_table_firstkey(q->factory_table);
+	while ( hash_table_nextkey(q->factory_table, factory_name, (void **) f) ) {
+		if ( !(*f)->queried && !(*f)->connected_workers ) {
```

Comment:
`if( !f->seen_at_catalog && f->connected_workes < 1) {`

Reply:
It communicates that connected_workers is not considered a boolean, but that it is a counter.